File: Linked List/Python/LC 876 - MIDDLE OF LINKED LIST.py
# LEET CODE 876 - Middle of the Linked List
# QUESTION LINK - https://leetcode.com/problems/middle-of-the-linked-list/

class Solution:
    def middleNode(self, head: Optional[ListNode]) -> Optional[ListNode]:
        slow = head
        fast = head

        while fast is not None and fast.next is not None:
            slow = slow.next
            fast = fast.next.next

        return slow

File: Linked List/Java/LC 876 - MIDDLE OF LINKED LIST.java
// LEET CODE 876 - Middle of the Linked List
// QUESTION LINK - https://leetcode.com/problems/middle-of-the-linked-list/

class Solution {
    public ListNode middleNode(ListNode head) {
        ListNode slow = head;
        ListNode fast = head;

        while (fast != null && fast.next != null) {
            slow = slow.next;
            fast = fast.next.next;
        }

        return slow;
    }
}
